rocAL/include/loaders/image_source_evaluator.h

```diff
@@ -41,10 +41,14 @@ enum class MaxSizeEvaluationPolicy {
 class ImageSourceEvaluator {
    public:
     ImageSourceEvaluatorStatus create(ReaderConfig reader_cfg, DecoderConfig decoder_cfg);
+    ImageSourceEvaluatorStatus create_numpy_reader(ReaderConfig reader_cfg);
     void find_max_dimension();
+    void find_max_numpy_dimensions();
     void set_size_evaluation_policy(MaxSizeEvaluationPolicy arg);
     size_t max_width();
     size_t max_height();
+    std::vector<size_t> max_numpy_dims() { return _max_numpy_dims; };
```

Comment:
Why is this added as image? It is supposed to be tensor

Reply:
I have extended the ImageSourceEvaluator class to run the numpy source evaluator for calculating max numpy dimensions and checking if the dtype and num_dims is the same for all numpy files